lib/events/events.go

```diff
@@ -144,16 +148,23 @@ func NewLogger() *Logger {
 
 func (l *Logger) Log(t EventType, data interface{}) {
 	l.mutex.Lock()
-	dl.Debugln("log", l.nextID, t, data)
-	l.nextID++
-	e := Event{
-		ID:   l.nextID,
-		Time: time.Now(),
-		Type: t,
-		Data: data,
-	}
+	dl.Debugln("log", l.nextGlobalID, t, data)
+	l.nextGlobalID++
+
+	now := time.Now()
+
 	for _, s := range l.subs {
 		if s.mask&t != 0 {
+			s.nextSubscriptionID++
+
+			e := Event{
```

Comment:
Sure, that could work. But I was imagining just a `ids []int` next to the `subs []*Subscription` on the `Logger`. Then the next ID for `subs[i]` is `ids[i]`.